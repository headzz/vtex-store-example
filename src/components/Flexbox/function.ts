const justify = {
  center: 'center',
  flexStart: 'flex-start',
  right: 'flex-end',
  spaceBetween: 'space-between'
}

type Allign = 'center' | 'right' | 'spaceBetween' | 'flexStart'

/**
 * @function
 * @name checkAlignContent
 * @param {Allign} alignments defines which alignment flex is going to use.
 * @description Creates flex structure for components
 */

export const checkAlignContent = (alignments: Allign): string => {
  return `justify-content: ${justify[alignments]}`
}
