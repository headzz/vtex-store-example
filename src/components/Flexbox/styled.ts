import styled from 'styled-components'
import { checkAlignContent } from './function'

type Props = {
  alignContent: 'center' | 'right' | 'spaceBetween' | 'flexStart'
  column?: boolean
}

export const FlexStyled = styled('div')<Props>`
  ${(props) => checkAlignContent(props.alignContent)};
  align-items: center;
  display: flex;
  flex-direction: ${(props) => (props.column ? 'column' : 'row')};
  height: 100%;
  width: 100%;
`
