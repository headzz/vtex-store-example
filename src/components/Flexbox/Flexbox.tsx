import { FlexStyled } from './styled'
import React from 'react'

type Props = {
  alignContent: 'center' | 'right' | 'spaceBetween' | 'flexStart'
  column?: boolean
}

/**
 * @component
 * @name Flexbox
 * @deion Gives flex display for components.
 * @namespace components components/Flexbox
 * @param {string} alignContent Justify content based on required parameter.
 * @param {element} children Component(s) that may appear inside the layout.
 * @param {boolean} column Changes flex type to column.
 *
 */

const Flexbox: React.FC<Props> = ({ children, column, alignContent }) => {
  return (
    <FlexStyled alignContent={alignContent} column={column}>
      {children}
    </FlexStyled>
  )
}

export default Flexbox
