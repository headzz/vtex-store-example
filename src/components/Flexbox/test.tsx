import { render, screen } from '@testing-library/react'

import Flexbox from './Flexbox'

describe('Flexbox', () => {
  it('should render component that represents display flex on page', () => {
    const { container } = render(
      <Flexbox alignContent="flexStart">
        <div>Teste</div>
      </Flexbox>
    )
    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild).toHaveStyle('justify-content: flex-start')
    expect(screen.getByText('Teste')).toBeInTheDocument()
  })

  it('should render component that represents display flex on page with content to the right', () => {
    const { container } = render(
      <Flexbox alignContent="right">
        <div>Teste</div>
      </Flexbox>
    )
    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild).toHaveStyle('justify-content: flex-end')
    expect(screen.getByText('Teste')).toBeInTheDocument()
  })

  it('should render component that represents display flex on page with content centered', () => {
    const { container } = render(
      <Flexbox alignContent="center">
        <div>Teste</div>
      </Flexbox>
    )
    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild).toHaveStyle('justify-content: center')
    expect(screen.getByText('Teste')).toBeInTheDocument()
  })

  it('should render component that represents display flex on page with space-between content', () => {
    const { container } = render(
      <Flexbox alignContent="spaceBetween">
        <div>Teste</div>
        <div>Teste2</div>
      </Flexbox>
    )
    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild).toHaveStyle('justify-content: space-between')
    expect(screen.getByText('Teste')).toBeInTheDocument()
  })
})
