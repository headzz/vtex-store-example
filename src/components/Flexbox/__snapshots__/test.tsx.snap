// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[
  `Flexbox should render component that represents display flex on page 1`
] = `
.c0 {
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: 100%;
  width: 100%;
}

<div
  class="c0"
>
  <div>
    Teste
  </div>
</div>
`

exports[
  `Flexbox should render component that represents display flex on page with content centered 1`
] = `
.c0 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: 100%;
  width: 100%;
}

<div
  class="c0"
>
  <div>
    Teste
  </div>
</div>
`

exports[
  `Flexbox should render component that represents display flex on page with content to the right 1`
] = `
.c0 {
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: 100%;
  width: 100%;
}

<div
  class="c0"
>
  <div>
    Teste
  </div>
</div>
`

exports[
  `Flexbox should render component that represents display flex on page with space-between content 1`
] = `
.c0 {
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: 100%;
  width: 100%;
}

<div
  class="c0"
>
  <div>
    Teste
  </div>
  <div>
    Teste2
  </div>
</div>
`
